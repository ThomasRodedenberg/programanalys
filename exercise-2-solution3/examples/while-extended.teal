fun check(a0: nonnull int, a1 : nonnull int, a2: nonnull int) = {
    return 1;
}

fun main () = {
    var v := 1;
    var w := 1;
    var x := 0;
    var y := 0;
    var z := 0;
    var n := null;

    v := z;
    while x < 10 {
        var a := 1;
        x := y;
        while x > y {
	   y := a;
	   a := z;
	   w := v;
	}
        z := n;
    }
    check(v, w, x);
    check(y, z, n);
}
