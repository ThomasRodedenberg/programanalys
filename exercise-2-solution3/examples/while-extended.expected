maybe: N 2 11 +
maybe: NC 14 10 ?+
maybe: NC 17 14 ??
NC 24 4 ++?
NC 25 4 ??0
